pennylane_lightning/src/algorithms/AdjointDiff.hpp

```diff
@@ -525,15 +642,16 @@ template <class T = double> class AdjointJacobian {
      * @param adj Indicate whether to take the adjoint of the operation.
      * @return T Generator scaling coefficient.
      */
-    inline T applyGenerator(StateVectorManaged<T> &sv,
-                            const std::string &op_name,
-                            const std::vector<size_t> &wires, const bool adj) {
+    inline auto applyGenerator(StateVectorManaged<T> &sv,
```

Comment:
Wouldn't pass-by-value and using `std::move` be a better practice here?

Reply:
Yea, Trevor is correct. We are not making any copies inside the method here, so copy-and-move would be extra work than reference only. The copy-and-move semantic assumes only works in those instances, and only when moving is cheap also.